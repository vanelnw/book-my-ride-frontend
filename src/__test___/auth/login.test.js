import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';

import Login from '../../components/auth/Login';

describe('Testing Login Component', () => {
  let signin;
  beforeAll(async () => {
    signin = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Login />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('Component should render correctly', () => {
    expect(signin).toBeTruthy();
  });

  it('It should match its snapshot', () => {
    expect(signin).toMatchSnapshot();
  });

  it('It should render the email input field', () => {
    const emailInput = waitFor(() => screen.getByLabelText('Email'));
    waitFor(() => expect(emailInput).toBeInTheDocument());
    waitFor(() => expect(emailInput.type).toBe('email'));
    waitFor(() => expect(emailInput.required).toBe(true));
  });

  it('It should render the password input field', () => {
    const passwordInput = waitFor(() => screen.getByLabelText('Password'));
    waitFor(() => expect(passwordInput).toBeInTheDocument());
    waitFor(() => expect(passwordInput.type).toBe('password'));
    waitFor(() => expect(passwordInput.required).toBe(true));
  });
});
