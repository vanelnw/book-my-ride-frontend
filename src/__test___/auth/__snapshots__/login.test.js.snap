// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login Component It should match its snapshot 1`] = `
<div
  className="loginPage"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="top"
    >
      <span>
        Don't have an account yet?Â 
        <a
          href="/register"
          onClick={[Function]}
        >
          Register Here
        </a>
      </span>
      <header>
        Login
      </header>
    </div>
    <div
      className="input-field"
    >
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </div>
    <div
      className="input-field"
    >
      <input
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <button
      className="submit"
      disabled={false}
      type="submit"
    >
      Login
    </button>
    <div
      className="two-col"
    >
      <div
        className="one"
      >
        <input
          id="check"
          name=""
          type="checkbox"
        />
        <label
          htmlFor="check"
        >
           Remember Me
        </label>
      </div>
      <div
        className="two"
      >
        <label>
          <a
            href="./LOGIN"
          >
            Forgot password?
          </a>
        </label>
      </div>
    </div>
  </form>
</div>
`;
